### DISCUSSION TITLE
2019-2022·相机算法集成·总结概述

### DISCUSSION THREAD

**Discussion by caibingcheng** [caibingcheng/comment#298] [Announcements]
https://imcbc.cn/202209/2019-2022-work-summary/

**Comment by penndu** [comment]
图片终于可以正常显示了啊！

**Comment by caibingcheng** [comment]
> @penndu
> 图片终于可以正常显示了啊！

我还是换成jpg图了

**Comment by ryan4yin** [comment]
之前也读过一遍，今天又读了第二遍，是我从未接触过的全新领域，不过仍然得出了一个很通用的结论：抽象（中间层）要取得一个平衡，这个平衡大概就是设计的艺术了。

就拿我的领域来说，容器编排领域曾经这么多工具，最后夺得第一的是谷歌开源的 Kubernetes，而曾经与  Kubernetes 同台竞技的其他竞争者，都渐渐无人问津了。
Kubernetes 成功最大的原因，就是它从谷歌几十年容器编排经验中提炼出的一套抽象模型，以及这套抽象带来的拓展能力与广阔的想象空间。

至于说「一些社会现实」，可能因为我还没在大厂呆过，说实话感觉并不明显。
我上家公司才 20 多人，整个运维团队就 2 到 3 个人，没啥内部斗争空间。
现在这家公司大家也更多的是合作者态度，领导也很 nice，给了我们很大的自由空间（这也是为什么我这两年的年终总结看着都很丰富的原因之一）。

**Comment by ryan4yin** [comment]
从我外行的角度看，feature/pipeline/node 这些名词与各种处理流程，感觉好像跟大数据有点像了哈哈，也是各种数据流的处理。

**Comment by caibingcheng** [comment]
> @ryan4yin
> 之前也读过一遍，今天又读了第二遍，是我从未接触过的全新领域，不过仍然得出了一个很通用的结论：抽象（中间层）要取得一个平衡，这个平衡大概就是设计的艺术了。
> 
> 就拿我的领域来说，容器编排领域曾经这么多工具，最后夺得第一的是谷歌开源的 Kubernetes，而曾经与  Kubernetes 同台竞技的其他竞争者，都渐渐无人问津了。
> Kubernetes 成功最大的原因，就是它从谷歌几十年容器编排经验中提炼出的一套抽象模型，以及这套抽象带来的拓展能力与广阔的想象空间。
> 
> 至于说「一些社会现实」，可能因为我还没在大厂呆过，说实话感觉并不明显。
> 我上家公司才 20 多人，整个运维团队就 2 到 3 个人，没啥内部斗争空间。
> 现在这家公司大家也更多的是合作者态度，领导也很 nice，给了我们很大的自由空间（这也是为什么我这两年的年终总结看着都很丰富的原因之一）。

关于「一些社会现实」，我有一部分原因大概是接触了太多别人的负能量了😅，所以就容易变得很“祥林嫂”。因为以前有一位同事特别特别喜欢和我吐槽，我也一直认真听，在他之前我对这些是没有什么感觉的，所以啊，还是**少接触一些别人的负面情绪**比较好，可能会被传染。不过还是看个人吧，因为我对此抵御能力不太好，所以容易被传染，还是磨练自己的心智更重要呀～
